src/bepasty/app.py

```diff
@@ -32,25 +34,68 @@ def __init__(self, app, prefix=''):
         self.app = app
         self.prefix = prefix
 
-    def __call__(self, environ, start_response):
+    def __call__(self, environ, start_response, set_script_name=True):
         if environ['PATH_INFO'].startswith(self.prefix):
             environ['PATH_INFO'] = environ['PATH_INFO'][len(self.prefix):]
-            environ['SCRIPT_NAME'] = self.prefix
+            if set_script_name:
+                environ['SCRIPT_NAME'] = self.prefix
             return self.app(environ, start_response)
         else:
             start_response('404', [('Content-Type', 'text/plain')])
             return ['This URL does not belong to the bepasty app.'.encode()]
```

Comment:
I guess this is all what we need from this PR and the rest should be done with / within werkzeug.

Reply:
And, this behaviour needs to be configured in the middleware `__init__`, not in `__call__`.